subsys/net/lib/lwm2m/lwm2m_message_handling.c

```diff
@@ -362,6 +364,7 @@ STATIC int build_msg_block_for_send(struct lwm2m_message *msg, uint16_t block_nu
 	} else {
 		/*  update block context */
 		msg->out.block_ctx->current = block_num * block_size_bytes;
+		msg->out.block_ctx->block_size = block_size;
```

Comment:
Same as https://github.com/zephyrproject-rtos/zephyr/pull/71323/files#r1559422341
```suggestion
		msg->out.block_ctx->block_size = convert_to_coap_block_size_enum(block_size);
```